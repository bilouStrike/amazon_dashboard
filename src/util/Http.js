import axios from 'axios';
import { BASE_API_URL } from 'constants/Common';

export default axios.create({
  baseURL: BASE_API_URL,
  headers: {
    'Content-Type': 'application/json',
  }
});
