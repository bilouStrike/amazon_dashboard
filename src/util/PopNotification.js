import { message } from "antd";

const showMessahe = (status, response) => {
    
    message.destroy();
    let config = {
        content:response,
        duration: 5,
        style: { padding:'120px', marginTop:'5%' }
    }

    if( status === 'success' ) {
        message.success(config);
        return;
    }
    message.error(config);
}
export default showMessahe;