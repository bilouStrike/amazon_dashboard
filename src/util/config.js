module.exports = {
  footerText: 'Copyright Amazon Tool Â© 2020',
}
