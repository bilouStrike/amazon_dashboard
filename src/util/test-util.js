// test-utils.js
import React from 'react'
import { render } from '@testing-library/react'
/*import { ThemeProvider } from 'my-ui-lib'
import { TranslationProvider } from 'my-i18n-lib'
import defaultStrings from 'i18n/en-x-default'*/
import { IntlProvider } from "react-intl";

const AllTheProviders = ({ children }) => {
  return (
    <IntlProvider locale="en">
        {children}
    </IntlProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }