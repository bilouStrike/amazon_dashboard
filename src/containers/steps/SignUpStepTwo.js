import React, {useState} from "react";
import { Button, Checkbox, Upload, Input, Form } from "antd";
import { UploadOutlined } from '@ant-design/icons';

import IntlMessages from "util/IntlMessages";
import { signUp } from 'services/auth';
import { randomInteger } from 'helpers';
import PopNotification from 'util/PopNotification';
import { BASE_API_URL } from 'constants/Common';

const SignUpStepTwo = ({stepOneValues, setStepOneValues, stepTwoValues, setStepTwoValues,  setStep}) => {

    const [load, setLoad] = useState(false);
    const prevStep = () => {
        setStepOneValues(stepOneValues);
        setStep(1);
    }

    const normFile = e => {
        console.log('Upload event:', e);
        if (Array.isArray(e)) {
          return e;
        }
        return e && e.fileList;
    };


    const fieldChange = async (values) => {
        const data = {};
        const name = values[0] && values[0].name[0];
        const value = values[0] && values[0].value;
        if( name && value ) {
            data[name] = value ;
            setStepTwoValues({...stepTwoValues, ...data })
        }
    }

    const onSignUp = async (values) => {
        const agencyId = randomInteger(9999, 9999999); // agencyId should generte from the server
        const userData = { 
          ...stepOneValues,
          ...values,
          'agencyId':agencyId,
          'companyId': null,
          roles:[{
            companyId: 0,
            name: "agency_owner",
            roleId: 1,
            type: "agency"}] 
        }
        setLoad(true);
        const { status, message } = await signUp(userData);
        setLoad(false);
        PopNotification(status, message)
    }

    return (
        <Form
            onFinish={onSignUp}
            name="basic"
            onFieldsChange={fieldChange}
            className="gx-signin-form gx-form-row0"
            initialValues={{ 
                agencyName: stepTwoValues.agencyName,
                phone: stepTwoValues.phone,
                address: stepTwoValues.address
            }}
        >
            <Form.Item rules={[
                    { required: true, message: 'Please input the agency name!'},
                    { pattern: new RegExp(/^[A-Za-z ]+$/), message: 'Agency Name should be alphabetic'}]
                  } 
                    name="agencyName"
                >
                  <Input placeholder="Agency name" className="amazon-tool-home-input"/>
            </Form.Item>

            <Form.Item rules={[
                    { required: true, message: 'Please input the agency name!'}]}
                    name="address"
                >
                  <Input placeholder="Address" className="amazon-tool-home-input"/>
            </Form.Item>

            <Form.Item rules={[
                    { required: true, message: 'Please input the agency phone!'},
                    { pattern: new RegExp(/^[+]*[(]{0,1}[0-9]{1,3}[)]{0,1}[-\s\./0-9]*$/g), message: 'Enter valid phone number'}]
                  } 
                    name="phone"
                >
                  <Input placeholder="Phone" className="amazon-tool-home-input"/>
            </Form.Item>

            <Form.Item
                name="Agency Logo"
                label="Upload"
                valuePropName="fileList"
                getValueFromEvent={normFile}
                extra="Logo format: JPG,PNG"
            >
                <Upload name="logo" action={`${BASE_API_URL}/upload`} listType='picture'> 
                    <Button>
                        <UploadOutlined /> Click to upload
                    </Button>
                </Upload>
            </Form.Item>

            <Form.Item>
                  <Button type="primary" className="gx-mb-0" style={{height:'48px'}} size="large" block htmlType="submit">
                    <IntlMessages id="app.userAuth.NextStepStartTrial"/>
                  </Button>
            </Form.Item>
        </Form>
    )
            
}
export default SignUpStepTwo;