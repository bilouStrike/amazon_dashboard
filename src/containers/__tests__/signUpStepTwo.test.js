import React from "react";
import "@testing-library/jest-dom/extend-expect";
import '../../util/testConfig';
import { render, fireEvent, cleanup } from '../../util/test-util';

import SignUpStepTwo from '../steps/SignUpStepTwo';

afterEach(cleanup);

it("should render signinstepTwo component", () => {
    
  const mockStepOneValues = {};
  const mockSetStepOneValues = jest.fn();
  const mockSetStep = jest.fn();
  const mockStepTwoValues = {};
  const mockSetStepTwoValues = jest.fn();
  const { asFragment } = render(
    <SignUpStepTwo 
        stepOneValues={mockStepOneValues}
        setStepOneValues={mockSetStepOneValues}
        setStep={mockSetStep}
        stepTwoValues={mockStepTwoValues}
        setStepTwoValues={mockSetStepTwoValues}
    />
    );
  expect(asFragment()).toMatchSnapshot();
});
