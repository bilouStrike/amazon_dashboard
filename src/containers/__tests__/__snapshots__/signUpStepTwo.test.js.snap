// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render signinstepOne component 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal gx-signin-form gx-form-row0"
    id="basic"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_FirstName"
              placeholder="First Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_LastName"
              placeholder="Last Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_email"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_username"
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_confirm"
              placeholder="Confirm password"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn gx-mb-0 ant-btn-primary ant-btn-lg ant-btn-block"
              style="height: 48px;"
              type="submit"
            >
              app.userAuth.NextStep
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`should render signinstepTwo component 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal gx-signin-form gx-form-row0"
    id="basic"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_agencyName"
              placeholder="Agency name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_address"
              placeholder="Address"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_phone"
              placeholder="Phone"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class=""
          for="basic_Agency Logo"
          title="Logo"
        >
          Logo
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <span
              class=""
            >
              <div
                class="ant-upload ant-upload-select ant-upload-select-picture"
              >
                <span
                  class="ant-upload"
                  role="button"
                  tabindex="0"
                >
                  <input
                    accept=""
                    id="basic_Agency Logo"
                    style="display: none;"
                    type="file"
                  />
                  <button
                    class="ant-btn"
                    type="button"
                  >
                    <span
                      aria-label="upload"
                      class="anticon anticon-upload"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="upload"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                    <span>
                       Click to upload
                    </span>
                  </button>
                </span>
              </div>
              <div
                class="ant-upload-list ant-upload-list-picture"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-form-item-extra"
        >
          Logo format: JPG,PNG
        </div>
      </div>
    </div>
    <a>
       Previous Step 
    </a>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn gx-mb-0 amz-top-margin-20 ant-btn-primary ant-btn-lg ant-btn-block"
              style="height: 48px;"
              type="submit"
            >
              app.userAuth.NextStepStartTrial
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
